csm/common/errors.py

```diff
@@ -119,31 +134,55 @@ class CsmPermissionDenied(CsmError):
     This error is raised for all cases when we don't have permissions
     """
 
-    def __init__(self, desc=None, message_id=None, message_args=None):
+    _desc = "Access to the requested resource is forbidden"
+
+    def __init__(self, desc=None, message_id=const.PERMISSION_DENIED_ERROR, message_args=None):
         super(CsmPermissionDenied, self).__init__(
-            CSM_INTERNAL_ERROR, desc,
+            CSM_OPERATION_NOT_PERMITTED, desc,
             message_id, message_args)
 
 
 class CsmResourceNotAvailable(CsmInternalError):
 
     """Describes issues when requested resource is not available"""
 
+    _desc = "Requested resource is not available"
+
+    def __init__(self, desc=None, message_id=const.RESOURCE_NOT_AVAILABLE, message_args=None):
+        super(CsmResourceNotAvailable, self).__init__(
+            desc, message_id, message_args)
 
 class CsmTypeError(CsmInternalError):
 
     """Issues related to incorrect type of argument/parameter, etc."""
 
+    _desc = "Incorrect type of argument/parameter"
+
+    def __init__(self, desc=None, message_id=const.TYPE_ERROR, message_args=None):
+        super(CsmTypeError, self).__init__(
+            desc, message_id, message_args)
 
 class CsmNotImplemented(CsmError):
 
     """This error represents HTTP 501 Not Implemented Error"""
 
+    _desc = "Not Implemented Error"
```

Comment:
Not Implemented Error --> Not Implemented

Reply:
For this error, message and message id will be similar
message = "Not Implemented"
message_id = "NotImplemented"
is it ok?